bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -650,8 +650,18 @@ impl Connection {
     }
 
     /// Attempts to flush any data previously buffered by a call to [send](`Self::poll_send`).
+    ///
+    /// poll_flush can only flush data that s2n-tls has already encrypted and
+    /// buffered for sending. poll_send may need to be called again to fully send
+    /// all data. See the [Usage Guide](https://github.com/aws/s2n-tls/blob/main/docs/usage-guide/topics/ch07-io.md)
+    /// for more details.
     pub fn poll_flush(&mut self) -> Poll<Result<&mut Self, Error>> {
```

Comment:
poll_send returns an error if it's called during renegotiation:
https://github.com/aws/s2n-tls/blob/9cd117f3d9a94912d5e8f2a61ffbfd31f11d2946/bindings/rust/s2n-tls/src/renegotiate.rs#L306-L313

Why is an error necessary for poll_send but not for poll_flush? Is it because poll_flush is guaranteed to no-op during renegotiate because the send buffer is guaranteed to be empty?

Reply:
> Is it because poll_flush is guaranteed to no-op during renegotiate because the send buffer is guaranteed to be empty?

Exactly. If there was data in the send buffer, that would have caused an error earlier and prevented renegotiation. Since you can't have existing data in the buffer and can't call poll_send to load more data into the buffer, poll_flush will always be a no-op during renegotiation.